pkg/vsphere/simulator/cluster_compute_resource.go

```diff
@@ -81,8 +81,8 @@ func CreateClusterComputeResource(f *Folder, name string, spec types.ClusterConf
 	cluster := &ClusterComputeResource{}
 	cluster.Name = name
 
-	pool := esx.ResourcePool
-	Map.PutEntity(cluster, Map.NewEntity(&pool))
+	pool := NewResourcePool()
```

Comment:
The root pool is already there by default:
```
% govc ls -l host/DC0_C0
/DC0/host/DC0_C0/DC0_C0_H0 (HostSystem)
/DC0/host/DC0_C0/DC0_C0_H1 (HostSystem)
/DC0/host/DC0_C0/DC0_C0_H2 (HostSystem)
/DC0/host/DC0_C0/Resources (ResourcePool)
```
This supports creating child pools:
```
% govc ls -l host/DC0_C0/Resources/*

% govc pool.create DC0_C0/Resources/child

% govc ls -l host/DC0_C0/Resources/*
/DC0/host/DC0_C0/Resources/child (ResourcePool)

% govc ls -l host/DC0_C0/Resources/child/*

% govc pool.create DC0_C0/Resources/child/grandkid

% govc ls -l host/DC0_C0/Resources/child/*
/DC0/host/DC0_C0/Resources/child/grandkid (ResourcePool)
```